tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause.bal

```diff
@@ -262,6 +267,74 @@ function testBreakWithinOnfail() {
     assertEquality("-> Before error thrown -> Error caught! Loop broke with digit: 1", str);
 }
 
+function testReturnWithinOnfail() returns string {
+    int i = 0;
+    string str = "";
+    while (i < 2) {
+        do {
+            str += " -> Before error thrown, ";
+            fail getError();
+        } on fail error e {
+            str += " -> Error caught at level #1";
+            fail getError();
+        }
+        i = i + 1;
+    } on fail error e {
+        str += " -> Error caught at levet #2";
+        return str;
+    }
+    return "should not reach here";
+}
+
+function testIntReturnWithinOnfail() returns int {
+    int i = 1;
+    while (i < 2) {
+        do {
+            fail getError();
+        } on fail error e {
+            fail getError();
+        }
+        i = i + 1;
+    } on fail error e {
+        return i;
+    }
+    return 0;
+}
+
+function testBreakWithinOnfailForOuterLoop() {
+    string str = "";
+    foreach int digit in 1...5 {
+        do {
+            if (digit > 3) {
+                fail getError();
+            }
+        } on fail error e {
+            str += "Loop broke with digit: " + digit.toString();
+            break;
```

Comment:
Shall we have a test case for continue as well?

Reply:
Refactored to test both continue and brake flows in b43693abd7969d50dcc6cfecb07ccffce07a870a